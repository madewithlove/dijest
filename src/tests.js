export {default as component} from './tests/component';
export {reducerTestFactory, default as reducer} from './tests/reducer';
export {default as saga} from './tests/saga';
