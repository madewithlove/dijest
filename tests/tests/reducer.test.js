import React from 'react';
import {fromJS} from 'immutable';
import {reducer, reducerTestFactory} from '../../dist/tests';

const dummyReducer = (state, action) => fromJS({state, action});

reducer('can test reducer', dummyReducer, {foo: 'bar'}, {type: 'FOO'});
reducer('can test reducer with result', dummyReducer, {foo: 'bar'}, {type: 'FOO'}, {state: {foo: 'bar'}, action: {type: 'FOO'}});

const tester = reducerTestFactory(dummyReducer);
const before = {foo: 'bar'};
const action = {type: 'FOO'};
const after = {state: {foo: 'bar'}, action: {type: 'FOO'}};

tester('can test reducer with factory', before, action);
tester('can test reducer with factory and result', before, action, after);
tester('can do multiple assertions', assert => {
    assert(before, action);
    assert(before, action, after);
});