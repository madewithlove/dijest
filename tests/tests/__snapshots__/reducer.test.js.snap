exports[`test can do multiple assertions 1`] = `
Object {
  "action": Object {
    "type": "FOO"
  },
  "state": Object {
    "foo": "bar"
  }
}
`;

exports[`test can test reducer 1`] = `
Object {
  "action": Object {
    "type": "FOO"
  },
  "state": Object {
    "foo": "bar"
  }
}
`;

exports[`test can test reducer with factory 1`] = `
Object {
  "action": Object {
    "type": "FOO"
  },
  "state": Object {
    "foo": "bar"
  }
}
`;
